File: agent/me/yuugiri/agent/MergedMap.java
package me.yuugiri.agent;

import java.util.List;
import java.util.Map;

public class MergedMap {

    public List<Map<String, String>> maps;

    public MergedMap(List<Map<String, String>> maps) {
        this.maps = maps;
    }

    public String get(String key) {
        for (Map<String, String> map : maps) {
            if (map.containsKey(key)) {
                return map.get(key);
            }
        }
        return null;
    }

    public String getWithFallback(String key, String fallback) {
        for (Map<String, String> map : maps) {
            if (map.containsKey(key)) {
                return map.get(key);
            }
        }
        return fallback;
    }
}

File: src/main/kotlin/me/yuugiri/fluiditygradle/utils/MergedMap.kt
package me.yuugiri.fluiditygradle.utils

class MergedMap<K, out V>(val maps: List<Map<K, V>>) {

    operator fun get(key: K): V? {
        maps.forEach {
            val value = it[key]
            if (value != null) return value
        }
        return null
    }
}